clippy_lints/src/methods/mod.rs

```diff
@@ -3286,6 +3291,26 @@ fn lint_single_char_push_string(cx: &LateContext<'_>, expr: &hir::Expr<'_>, args
     }
 }
 
+/// lint for length-1 `str`s as argument for `insert_str`
+fn lint_single_char_insert_string(cx: &LateContext<'_>, expr: &hir::Expr<'_>, args: &[hir::Expr<'_>]) {
+    let mut applicability = Applicability::MachineApplicable;
+    if let Some(extension_string) = get_hint_if_single_char_arg(cx, &args[2], &mut applicability) {
+        let base_string_snippet =
+            snippet_with_applicability(cx, args[0].span.source_callsite(), "_", &mut applicability);
+        let pos_arg = snippet(cx, args[1].span, "..");
```

Comment:
This should probably also be `snippet_with_applicability`
```suggestion
        let pos_arg = snippet_with_applicability(cx, args[1].span, "..", &mut applicability);
```